### PR TITLE
Remove viewport workaround in CarPlay

### PR DESCRIPTION
**Pull request by frederoni**
Now that https://github.com/mapbox/mapbox-gl-native/pull/14664 gained traction, I took the chance to remove the hacky workaround for the viewport in CarPlay and the NavigationMapView.

This PR is blocked by the PR mentioned above, and a new release of the Maps SDK. As well as some cleanup work and testing.

This PR will unblock #1139

Fixes #2134.

cc @mapbox/navigation-ios @astojilj

### CODE REVIEW

**Comment by astojilj** [comment]
@frederoni Thanks. Looks good - simpler, though I'm not familiar with the code to properly review it.
It would be interesting to see how it affects performance - the viewport area should be now 2 times smaller:
from the screenshot looks like it was (1.5 * width + 100pixels) x (1.5 * height + 100pixels), width and height and 100px are in logical coordinates.

100 pixels comes from [CollisionIndex viewportPadding](https://github.com/mapbox/mapbox-gl-native/blob/2cfdd1a2f7042650538d24058fc28ad092739073/src/mbgl/text/collision_index.cpp#L25) that we could also try to tweak.

**Comment by 1ec5** [comment]
Awesome – so glad this is becoming a reality!

> 

I’m not sure why, but the vanishing point is offset slightly too far to the right, or the puck is offset slightly too far to the left.

**Comment by frederoni** [comment]
> I’m not sure why, but the vanishing point is offset slightly too far to the right, or the puck is offset slightly too far to the left.

Yea, the userAnchorPoint doesn't exactly match the safeArea’s centerY so there is a small discrepancy that'll need to be addressed before this PR can land. Good pixelhunting 👁

**Comment by 1ec5** [comment]
mapbox/mapbox-gl-native#14664 is shipping today as part of map SDK v5.1.0. Since the bug fix isn’t shipping as a semver-major change, the workaround needs to be removed to avoid overcorrecting the vanishing point.

**Comment by 1ec5** [comment]
#2143 prevents this SDK from working with v5.1.0 for the time being, but we need to remove that restriction and depend on v5.1._x_ in short order to avoid blocking various applications from upgrading the map SDK.

File: MapboxNavigation/NavigationMapView.swift [hunk 1]
```diff
@@ -337,10 +337,7 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         if tracksUserCourse {
             let newCamera = camera ?? MGLMapCamera(lookingAtCenter: location.coordinate, altitude: altitude, pitch: 45, heading: location.course)
             let function: CAMediaTimingFunction? = animated ? CAMediaTimingFunction(name: .linear) : nil
-            let point = userAnchorPoint
-            let padding = UIEdgeInsets(top: point.y, left: point.x, bottom: bounds.height - point.y, right: bounds.width - point.x)
-            // Omit padding when https://github.com/mapbox/mapbox-gl-native/pull/14081 has landed
```
Comment (1ec5) [inline MapboxNavigation/NavigationMapView.swift]: This comment was a bit misleading: the actual workarounds were the paddings added in `showcase(_:padding:animated:)` and `fit(to:facing:padding:animated:)`, not the one here. The edge padding here is necessary in order to account for the puck’s uncentered location even within the content frame.

/ref https://github.com/mapbox/mapbox-navigation-ios/issues/2145#issuecomment-503800503

**Comment by 1ec5** [comment]
These assertions are failing and crashing on iOS 12 for some odd reason:

https://github.com/mapbox/mapbox-navigation-ios/blob/db19ffe8b104ed25475916e95267a66da71da38d/MapboxCoreNavigationTests/NavigationEventsManagerTests.swift#L55-L57

**Pull request merged**